packages/server/src/controllers/Controller.js

```diff
@@ -533,3 +475,54 @@ export class Controller {
 EventEmitter.mixin(Controller.prototype)
 
 const inheritanceMap = new WeakMap()
+
+const setupHandlerFromObject = ({
+  handler,
+  action,
+  authorize,
+  parameters,
+  returns,
+  scope,
+  transacted
+}) => {
+  handler.authorize = authorize
+  handler.transacted = transacted
+
+  if (action) {
+    const [verb, path] = asArray(action)
+    handler.verb = verb
+    handler.path = path
+  }
+
+  if (parameters) {
+    const [_parameters, options] = parameters
+    const hasOptions = isArray(_parameters)
+    handler.parameters = hasOptions ? _parameters : parameters
+
+    // If validation options are provided, expose them through
+    // `handler.options.parameters`, see ControllerAction
+    if (hasOptions) {
+      handler.options = handler.options || {}
+      handler.options.parameters = options
+    }
+  }
+
+  if (returns) {
+    const [_returns, options] = asArray(returns)
+    handler.returns = _returns
+
+    // If validation options are provided, expose them through
+    // `handler.options.returns`, see ControllerAction
+    if (options) {
+      handler.options = handler.options || {}
+      handler.options.returns = options
+    }
+  }
+
+  if (scope) {
+    const handlerScope = handler.scope = handler.scope || []
+    handlerScope.push(...asArray(scope))
```

Comment:
I am wondering… Will this `handler` function from the object ever already have a scope property? I don't think so? And maybe we shouldn't even support that anyway, since it would mean somebody is mixing decorators and object declaration. In that case, this can be simplified down to this, outside of the `if` switch:

```js
handler.scope = scope
```

Reply:
Since scope can be string or array of string, we need to use `asArray`.  Since `asArray` will return an empty array if `scope` is undefined, it probably makes sense to keep the `if` switch:

```js
  if (scope) {
    handler.scope = asArray(scope)
  }
```

Reply:
Why do we need to use `asArray`? If it's a string, it can stay being a string on `handler.scope`, no?